exp/addons/controllers/clusterresourceset_controller.go

```diff
@@ -357,6 +357,23 @@ func (r *ClusterResourceSetReconciler) clusterToClusterResourceSet(o handler.Map
 		return nil
 	}
 
+	if !cluster.DeletionTimestamp.IsZero() {
```

Comment:
2 things:
1. This mapping function should not have any side effects (i.e. deleting things)
1. This has the potential to miss deleting bindings if the cluster is deleted but the controller doesn't see it. I would recommend just enqueuing the CRS, and in the reconcile method, if you can't find the cluster, delete its bindings.

Reply:
> This has the potential to miss deleting bindings if the cluster is deleted but the controller doesn't see it (offline during deletion). I would recommend just enqueuing the CRS, and in the reconcile method, if you can't find the cluster, delete its bindings.

When we reconcile CRS, if a cluster is deleted, it won't match with its label. So there is no way to know which clusters are deleted. One thing we can do is to list CRSBindings at each reconcile and check if their cluster still exist.

Reply:
Would it make sense to have a "controller" for ClusterResourceSetBindings that could handle the deletion.

Then it would remove the intermediate step of trying to map Clusters to ClusterResourceSets in order to clean up ClusterResourceSetBindings and it should be possible to filter out CRSBindings based on the ownerRefs on deletion for either a Cluster or a ClusterResourceSet

Reply:
A separate controller would probably be easier

Reply:
A separate controller makes sense but we will still need the Clusters to ClusterResourceSets mapping for applying CRS to newly created clusters.

Reply:
What about deleteting `ClusterResourceSetBindings` as part of `ClusterController.reconcileDelete`?

Reply:
We may want to avoid touching core controllers for features. Is there any issues you see doing it in the CRSBinding controller?

Reply:
I don't see the issues
I was not aware of this constraint